projects/ngrx-auto-entity-service/src/lib/entity.service.ts

```diff
@@ -48,25 +48,58 @@ export class EntityService implements IAutoEntityService<any> {
     return this.buildUrl('create', entityInfo, criteria).pipe(switchMap(url => this.http.post<any>(url, entity)));
   }
 
+  createMany(entityInfo: IEntityInfo, entities: any[], criteria?: any, originalEntities?: any[]): Observable<any[]> {
+    return this.buildUrl('createMany', entityInfo, criteria).pipe(switchMap(url => this.http.post<any>(url, entities)));
+  }
+
   update(entityInfo: IEntityInfo, entity: any, criteria?: EntityCriteria, originalEntity?: any): Observable<any> {
     return this.buildUrl('update', entityInfo, criteria, getKeyFromModel(entityInfo.modelType, entity)).pipe(
       switchMap(url => this.http.patch<any>(url, entity))
     );
   }
 
+  updateMany(entityInfo: IEntityInfo, entities: any[], criteria?: any, originalEntities?: any[]): Observable<any[]> {
+    return this.buildUrl('updateMany', entityInfo, criteria).pipe(switchMap(url => this.http.patch<any>(url, entities)));
+  }
+
   replace(entityInfo: IEntityInfo, entity: any, criteria?: EntityCriteria, originalEntity?: any): Observable<any> {
     return this.buildUrl('replace', entityInfo, criteria, getKeyFromModel(entityInfo.modelType, entity)).pipe(
       switchMap(url => this.http.put<any>(url, entity))
     );
   }
 
+  replaceMany(entityInfo: IEntityInfo, entities: any[], criteria?: any, originalEntities?: any[]): Observable<any[]> {
+    return this.buildUrl('replaceMany', entityInfo, criteria).pipe(switchMap(url => this.http.put<any>(url, entities)));
+  }
+
   delete(entityInfo: IEntityInfo, entity: any, criteria?: EntityCriteria, originalEntity?: any): Observable<any> {
     return this.buildUrl('delete', entityInfo, criteria, getKeyFromModel(entityInfo.modelType, entity)).pipe(
-      switchMap(url => this.http.delete<any>(url, entity).pipe(map(() => entity)))
+      switchMap(url =>
+        this.http.delete<any>(url, {
+          body: entity
+        })
+      ),
+      map(() => entity)
+    );
+  }
+
+  deleteMany(entityInfo: IEntityInfo, entities: any[], criteria?: any, originalEntities?: any[]): Observable<any[]> {
+    return this.buildUrl('deleteMany', entityInfo, criteria).pipe(
+      switchMap(url => this.http.delete<any>(url, { body: entities })),
+      map(() => entities)
     );
   }
 
   deleteByKey(entityInfo: IEntityInfo, key: EntityIdentity, criteria?: EntityCriteria): Observable<EntityIdentity> {
     return this.buildUrl('deleteByKey', entityInfo, criteria, key).pipe(switchMap(url => this.http.delete<any>(url).pipe(map(() => key))));
   }
+
+  deleteManyByKeys(entityInfo: IEntityInfo, keys: EntityIdentity[], criteria?: any): Observable<any[]> {
+    // TODO: Is this the best way to handle this case?
+    //  This seems like a somewhat odd REST API pattern.
+    return this.buildUrl('deleteManyByKeys', entityInfo, criteria).pipe(
+      switchMap(url => this.http.delete<any>(url, { body: { keys } })),
+      map(() => keys) // Will auto entity handle this correctly and remove the entities from state?
```

Comment:
Since this isn't fully understood, does it need to be added?

Reply:
Most of these methods are were asked for by the Timebug team to allow the use of bulk endpoints and reduce the api calls. I figured for now this would work just fine since the generic service would also allow for different bulk delete by key approaches, such as with query params. Also, I did some digging into the deleteManyByKeys and it looks like it should work fine, so I can remove the second comment.

Reply:
Sounds good. If you can update or remove the comments we can merge this.